### ISSUE TITLE
Documentation "Number parsing is not properly implemented yet"

### ISSUE THREAD

**Issue by dadonzg** [python-babel/babel#692]
Good Morning,

First thank you for this very helpful library,

I have a question related to the documentation,

On https://github.com/python-babel/babel/blob/master/docs/numbers.rst#parsing-numbers

You are saying:

> Number parsing is not properly implemented yet

Is it related to only `parse_number` or also to `parse_decimal` ?

Thank you,

**Comment by bradleypmartin** [comment]
Hey @dadonzg, thanks for your input - I'm going to check this out a little bit today.

I'm _guessing_ this has to do with the `parse_number` function not being implemented, but if so, I agree - we could make that a little more explicit/clear in the docs.

**Comment by bradleypmartin** [comment]
Hey once more, dadonzg! Just wanted to update you with what I'd found and done today. I checked out the source and did some testing of my own with both the `parse_decimal` and `parse_number` functions (and some of those tests are reflected in the PR now linked here: https://github.com/python-babel/babel/pull/708)

The `parse_number` logic is indeed implemented, but isn't built out to the same extent that `parse_decimal` is - particularly in regard to handling special cases (including the 'ru' locale space parsing issue that I included in the docs PR above).

So after investigation, I believe your question definitely relates at least _more_ to `parse_number`, and I've tried to clear that up in the PR.
